libraries/preferences/impl/src/main/kotlin/io/element/android/libraries/preferences/impl/store/SessionPreferencesModule.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.libraries.preferences.impl.store
+
+import com.squareup.anvil.annotations.ContributesTo
+import dagger.Module
+import dagger.Provides
+import io.element.android.features.preferences.api.store.SessionPreferencesStore
+import io.element.android.libraries.di.SessionScope
+import io.element.android.libraries.di.SingleIn
+import io.element.android.libraries.di.annotations.SessionCoroutineScope
+import io.element.android.libraries.matrix.api.user.CurrentSessionIdHolder
+import kotlinx.coroutines.CoroutineScope
+
+@Module
+@ContributesTo(SessionScope::class)
+object SessionPreferencesModule {
+    @Provides
+    @SingleIn(SessionScope::class)
```

Comment:
I think `@SingleIn(SessionScope::class)` is not necessary here. `defaultSessionPreferencesStoreFactory` is already handling the concurrency access.

Reply:
True, I was just keeping it there in case we can solve the underlying issue and remove `DefaultSessionPreferencesStoreFactory`. But it doesn't do anything at the moment, so we can remove it.